test/modules/TestResultCoordinator/storage/TestOperationResultStorage.cs

```diff
@@ -1,49 +1,39 @@
+        static readonly ILogger Logger = Microsoft.Azure.Devices.Edge.ModuleUtil.ModuleUtil.CreateLogger(nameof(TestOperationResultStorage));
+        private Dictionary<string, ISequentialStore<TestOperationResult>> resultStores;
 
-        public static async Task InitAsync(string storagePath, ISystemEnvironment systemEnvironment, bool optimizeForPerformance, List<string> resultSources)
+        public static async Task<TestOperationResultStorage> Create(IStoreProvider storeProvider, List<string> resultSources)
         {
-            StoreProvider storeProvider;
-            Preconditions.CheckNotNull(systemEnvironment);
-            Preconditions.CheckNotNull(optimizeForPerformance);
-            Preconditions.CheckNotNull(resultSources);
-            try
-            {
-                IDbStoreProvider dbStoreprovider = DbStoreProvider.Create(
-                    new RocksDbOptionsProvider(systemEnvironment, optimizeForPerformance, Option.None<ulong>()),
-                    GetStoragePath(storagePath),
-                    resultSources);
-
-                storeProvider = new StoreProvider(dbStoreprovider);
-            }
-            catch (Exception ex) when (!ExceptionEx.IsFatal(ex))
+            Preconditions.CheckNotNull(storeProvider, nameof(storeProvider));
+            Dictionary<string, ISequentialStore<TestOperationResult>> resultSourcesToStores = new Dictionary<string, ISequentialStore<TestOperationResult>>();
+            foreach (string resultSource in resultSources)
             {
-                Logger.LogError(ex, "Error creating RocksDB store. Falling back to in-memory store.");
-                storeProvider = new StoreProvider(new InMemoryDbStoreProvider());
+                resultSourcesToStores.Add(resultSource, await storeProvider.GetSequentialStore<TestOperationResult>(resultSource));
             }
 
-            resultStores = await InitializeStoresAsync(storeProvider, resultSources);
+            return new TestOperationResultStorage(resultSourcesToStores);
+        }
+
+        private TestOperationResultStorage(Dictionary<string, ISequentialStore<TestOperationResult>> resultStores)
+        {
+            this.resultStores = resultStores;
         }
 
-        public static ISequentialStore<TestOperationResult> GetStoreFromSource(string source)
+        public ISequentialStore<TestOperationResult> GetStoreFromSource(string source)
         {
-            return resultStores.TryGetValue(source, out ISequentialStore<TestOperationResult> store) ? store : throw new InvalidDataException($"Source {source} not found.");
+            return this.resultStores.TryGetValue(source, out ISequentialStore<TestOperationResult> store) ? store : throw new InvalidDataException($"Source {source} not found.");
         }
 
         static async Task<Dictionary<string, ISequentialStore<TestOperationResult>>> InitializeStoresAsync(StoreProvider storeProvider, List<string> resultSources)
```

Comment:
does this method still use?

Reply:
does GetStoragePath method still use?

Reply:
InitializeStoresAsync and GetStoragePath were not used. Removed them.